framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,241 @@
+                        false,
+                        "Whether to do Laplacian mesh smoothing on the generated triangles.");
+  params.addParam<std::vector<MeshGeneratorName>>(
+      "holes", std::vector<MeshGeneratorName>(), "The MeshGenerators that define mesh holes.");
+  params.addParam<std::vector<bool>>(
+      "stitch_holes", std::vector<bool>(), "Whether to stitch to the mesh defining each hole.");
+  params.addParam<std::vector<unsigned int>>("interpolate_holes",
+                                             std::vector<unsigned int>(),
+                                             "How many nodes to use per hole boundary segment.");
+  params.addParam<std::vector<bool>>("refine_holes",
+                                     std::vector<bool>(),
+                                     "Whether to allow automatically refining each hole boundary.");
+  params.addParam<Real>("desired_area", std::numeric_limits<Real>::max(), "Desired triangle area.");
+  params.addParam<std::string>("desired_area_func",
+                               std::string(),
+                               "Function specifying desired triangle area as a function of x,y.");
+  params.addParam<MooseEnum>(
+      "algorithm",
+      algorithm,
+      "Control the use of binary search for the nodes of the stitched surfaces.");
+  params.addParam<bool>(
+      "verbose_stitching", false, "Whether mesh stitching should have verbose output.");
+
+  params.addClassDescription("Triangulates meshes within boundaries defined by input meshes.");
+
+  return params;
+}
+
+Poly2TriMeshGenerator::Poly2TriMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _bdy_ptr(getMesh("boundary")),
+    _interpolate_bdy(getParam<unsigned int>("interpolate_boundary")),
+    _refine_bdy(getParam<bool>("refine_boundary")),
+    _smooth_tri(getParam<bool>("smooth_triangulation")),
+    _hole_ptrs(getMeshes("holes")),
+    _stitch_holes(getParam<std::vector<bool>>("stitch_holes")),
+    _interpolate_holes(getParam<std::vector<unsigned int>>("interpolate_holes")),
+    _refine_holes(getParam<std::vector<bool>>("refine_holes")),
+    _desired_area(getParam<Real>("desired_area")),
+    _desired_area_func(getParam<std::string>("desired_area_func")),
+    _algorithm(parameters.get<MooseEnum>("algorithm")),
+    _verbose_stitching(parameters.get<bool>("verbose_stitching"))
+{
+}
+
+std::unique_ptr<MeshBase>
+Poly2TriMeshGenerator::generate()
+{
+  // We put the boundary mesh in a local pointer
+  std::unique_ptr<UnstructuredMesh> mesh = dynamic_pointer_cast<UnstructuredMesh>(_bdy_ptr);
+
+  if (!_stitch_holes.empty() && _stitch_holes.size() != _hole_ptrs.size())
+    paramError("stitch_holes", "Need one stitch_holes entry per hole, if specified.");
+
+  if (!_interpolate_holes.empty() && _interpolate_holes.size() != _hole_ptrs.size())
+    paramError("interpolate_holes", "Need one interpolate_holes entry per hole, if specified.");
```

Comment:
You should be able to check both of these in the constructor

Reply:
Moved